src/ActionPanel.js

```diff
@@ -0,0 +1,135 @@
+import { ReactComponent as ReplaceIcon } from './svg/replace.svg';
+import './ActionPanel.scss';
+
+type ActionPanelProps = {
+    focusedOptionName: ?string,
+    selectedCommandName: ?string,
+    program: Program,
+    pressedStepIndex: number,
+    intl: any,
+    onDelete: (index: number) => void,
+    onReplace: (index: number) => void,
+    onMoveToPreviousStep: (index: number) => void,
+    onMoveToNextStep: (index: number) => void
+};
+
+class ActionPanel extends React.Component<ActionPanelProps, {}> {
+    actionPanelRef: { current: null | HTMLDivElement };
+
+    constructor(props) {
+        super(props);
+        this.actionPanelRef = React.createRef();
+    }
+
+    makeStepInfoMessage() {
+        const currentStepName = this.props.program[this.props.pressedStepIndex];
+
+        let ariaLabelObj = {
+            'stepNumber': this.props.pressedStepIndex + 1,
+            'stepName': this.props.intl.formatMessage({id:`CommandInfo.${currentStepName}`}),
+            'selectedCommandName': '',
+            'previousStepInfo': '',
+            'nextStepInfo': ''
+        };
+
+        if (this.props.selectedCommandName) {
+            ariaLabelObj['selectedCommandName'] =
+                this.props.intl.formatMessage(
+                    { id: 'ActionPanel.selectedCommandName' },
+                    { selectedCommandName: this.props.selectedCommandName }
+                );
+        }
+
+        if (this.props.pressedStepIndex > 0) {
+            const prevStepName = this.props.program[this.props.pressedStepIndex - 1];
+            ariaLabelObj['previousStepInfo'] =
+                this.props.intl.formatMessage(
+                    { id: `CommandInfo.previousStep.${prevStepName}`},
+                    { previousStepNumber: this.props.pressedStepIndex }
+                );
+        }
+
+        if (this.props.pressedStepIndex < (this.props.program.length - 1)) {
+            const nextStepName = this.props.program[this.props.pressedStepIndex + 1];
+            ariaLabelObj['nextStepInfo'] =
+                this.props.intl.formatMessage(
+                    { id: `CommandInfo.nextStep.${nextStepName}`},
+                    { nextStepNumber: this.props.pressedStepIndex + 2}
+                );
+        }
+
+        return ariaLabelObj;
+    }
+
+    render() {
+        const stepInfoMessage = this.makeStepInfoMessage();
+        return (
+            <div
+                id='ActionPanel'
+                className={'ActionPanel__panel'}
+                data-actionpanelgroup={true}
+                ref={this.actionPanelRef}>
+                <AriaDisablingButton
+                    name='deleteCurrentStep'
+                    disabled={false}
+                    aria-label={this.props.intl.formatMessage({id:'ActionPanel.action.delete'}, stepInfoMessage)}
+                    className='ActionPanel__action-buttons'
+                    onClick={()=>{this.props.onDelete(this.props.pressedStepIndex)}}>
```

Comment:
Change the `onClick` handlers for the action panel buttons to be consistent with our other components:

    onClick={this.handleClickDelete}

And implement `handle...` click handlers for each of the buttons.